source/renderer/app/features/menu-updater/makeReactionCallback.spec.js

```diff
@@ -0,0 +1,212 @@
+// @flow
+
+import { RouterStore } from 'mobx-react-router';
+import { WalletSettingsStateEnum } from '../../../../common/ipc/api';
+import { REDEEM_ITN_REWARDS_STEPS } from '../../config/stakingConfig';
+import type { RebuildApplicationMenu } from '../../ipc/rebuild-application-menu';
+import AppStore from '../../stores/AppStore';
+import ProfileStore from '../../stores/ProfileStore';
+import UiDialogsStore from '../../stores/UiDialogsStore';
+import makeReactionCallback from './makeReactionCallback';
+import { ROUTES } from '../../routes-config';
+import type { MakeReactionCallbackArgs } from './types';
+
+const makeApp = ({ activeDialog = false } = {}): AppStore =>
+  ({ activeDialog: activeDialog ? 'SOME_DIALOG' : null }: any);
+const makeProfile = ({
+  currentLocaleCallback = () => {},
+  areTermsOfUseAccepted = false,
+} = {}): ProfileStore =>
+  ({
+    get currentLocale() {
+      currentLocaleCallback();
+      return 'en';
+    },
+    areTermsOfUseAccepted,
+  }: any);
+const makeRouter = ({ pathname = ROUTES.WALLETS.ROOT } = {}): RouterStore =>
+  ({ location: { pathname } }: any);
+const makeStaking = ({ redeemOpen = false } = {}): RouterStore =>
+  ({
+    redeemStep: redeemOpen ? REDEEM_ITN_REWARDS_STEPS.CONFIGURATION : null,
+  }: any);
+const makeUiDialogs = ({ activeDialog = false } = {}): UiDialogsStore =>
+  ({
+    activeDialog: activeDialog ? () => {} : null,
+  }: any);
+const makeRebuildApplicationMenu = ({
+  send = () => {},
+} = {}): RebuildApplicationMenu => ({ send }: any);
+
+const defaultArgs: MakeReactionCallbackArgs = {
+  stores: {
+    app: makeApp(),
+    profile: makeProfile(),
+    router: makeRouter(),
+    staking: makeStaking(),
+    uiDialogs: makeUiDialogs(),
+  },
+  rebuildApplicationMenu: makeRebuildApplicationMenu(),
+};
+
+describe('MenuUpdater feature/makeReactionCallback', () => {
+  it('sends rebuildApplicationMenu IPC channel message', () => {
+    const send = jest.fn();
+    makeReactionCallback({
+      ...defaultArgs,
+      rebuildApplicationMenu: makeRebuildApplicationMenu({ send }),
+    })();
+
+    expect(send).toHaveBeenCalledTimes(1);
+    expect(send).toHaveBeenNthCalledWith(1, {
+      isNavigationEnabled: expect.any(Boolean),
+      walletSettingsState: expect.stringMatching(
+        new RegExp(
+          `(${WalletSettingsStateEnum.enabled}|${WalletSettingsStateEnum.disabled}|${WalletSettingsStateEnum.hidden})`
```

Comment:
Should we be more explicit on test assertion? Does it means that you are expecting any of these values? If so, you could refactor to use `test.each` and supply different params and outcomes.

Reply:
I was not aware of existence of `test.each`, it is cool :+1: However it is not the intension of this test case to test all of those values as all of them are tested bellow. My intention was to have a basic test case for ensuring the ipc channel communication is triggered by this functionality. I changed the `walletSettingsState` value to expect any string